src/Mvc/test/WebSites/BasicWebSite/StartupWithCustomValueProvider.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using BasicWebSite.ValueProviders;
+
+namespace BasicWebSite;
+
+public class StartupWithCustomValueProvider
```

Comment:
You don't really need this file. Since you can use the `Startup` class instead and update your unit test to include the ValueProviderFactory. eg:

``` c#
factory = factory.WithWebHostBuilder(b
            => b.ConfigureTestServices(serviceCollection => _serviceCollection = serviceCollection.Configure<MvcOptions>(o => o.ValueProviderFactories.Add(new CustomValueProviderFactory()))));
```

Reply:
However, I am ok to keep it if nobody is against it.